src/components/App/App.jsx

```diff
@@ -0,0 +1,116 @@
+import Navbar from "../Navbar/Navbar"
+import Home from "../Home/Home"
+import Profile from "../Profile/Profile"
+import Recipes from "../Recipes/Recipes"
+import Pantry from "../Pantry/Pantry"
+import ShoppingCart from "../ShoppingCart/ShoppingCart"
+import './App.css';
+
+import { BrowserRouter, Routes, Route } from "react-router-dom"
+import { useState, useEffect } from "react"
+
+export default function App() {
+
+  var basicProducts = [
+    {name: "apple",
+    quantity: 2},
+    {name: "orange",
+    quantity: 5},
+    {name: "grape",
+    quantity: 8},
+    {name: "bacon",
+    quantity: 3},
+    {name: "milk",
+    quantity: 2},
+    {name: "pasta",
+    quantity: 1}
+  ]
+
+  var basicProductForm = {
+    name: "",
+    quantity: ""
+  }
+
+const [products, setProducts] = useState(basicProducts)
+const [productForm, setProductForm] = useState(basicProductForm)
+
+const handleProductQuantity = (productName, operation) => {
+  // find index of productName
+  let itemIndex = products.findIndex(item => item.name === productName)
+
+  let newProducts = []
+  for (let i = 0; i < products.length; i++) {
+    newProducts.push(products[i])
+  }
```

Comment:
Why are you creating a new list? Can you just use the original products list?
If you do need a copy, you can probably do this more concisely with something like `let new products = [...products];`

Reply:
I need to create a deep copy of products each time I edit it since state variables can't be modified directly (instead I use the set function). I was hesitant to use `let new products = [...products];` since it is only a shallow copy, but I think for products (since it's only one dimensional) a shallow copy should work